tfhe/src/high_level_api/integers/types/compact.rs

```diff
@@ -90,3 +92,273 @@ where
+            &|ct: &mut Ct| ct.list.ct_list.pbs_order = PBSOrder::BootstrapKeyswitch,
+            &|ct: &mut Ct| {
+                *ct.list.ct_list.ct_list.get_mut_ciphertext_modulus() =
+                    CiphertextModulus::try_new_power_of_2(1).unwrap();
+            },
+            &|ct: &mut Ct| {
+                ct.list.ct_list.ct_list.get_mut_container().pop();
+            },
+            &|ct: &mut Ct| ct.list.ct_list.ct_list.get_mut_container().push(0),
+        ];
+
+        for breaker in breakers2 {
+            let mut ct_clone = ct.clone();
+
+            breaker(&mut ct_clone);
+
+            assert!(!ct_clone.is_conformant(&params));
+        }
+    }
+
+    #[test]
+    fn test_valid_generic_compact_integer() {
+        let config = ConfigBuilder::all_disabled()
+            .enable_default_integers()
+            .build();
+
+        let (client_key, server_key) = generate_keys(config);
+
+        let public_key = CompactPublicKey::new(&client_key);
+
+        set_server_key(server_key);
+
+        let ct = CompactFheUint8::try_encrypt(0, &public_key).unwrap();
+
+        assert!(ct.is_conformant(&PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()));
+
+        let mut rng = thread_rng();
+
+        for _ in 0..10 {
+            let mut ct_clone = ct.clone();
+
+            ct_clone
+                .list
+                .ct_list
+                .ct_list
+                .get_mut_container()
+                .fill_with(|| rng.gen::<u64>());
+
+            assert!(ct_clone.is_conformant(&PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()));
+
+            let mut ct_clone_expanded = ct_clone.expand();
+
+            ct_clone_expanded += &ct_clone_expanded.clone();
+        }
+    }
+
+    #[test]
+    fn test_valid_generic_compact_integer_list() {
+        let config = ConfigBuilder::all_disabled()
+            .enable_default_integers()
+            .build();
+
+        let (client_key, server_key) = generate_keys(config);
+
+        let public_key = CompactPublicKey::new(&client_key);
+
+        set_server_key(server_key);
+
+        let ct = CompactFheUint8List::try_encrypt(&[0, 1], &public_key).unwrap();
+
+        let params = (
+            PARAM_MESSAGE_2_CARRY_2_KS_PBS.into(),
+            ListSizeConstraint {
+                min_inclusive_group_count: 1,
+                max_inclusive_group_count: 1,
+                group_size: 2,
+            },
+        );
+
+        assert!(ct.is_conformant(&params));
+
+        let mut rng = thread_rng();
+
+        for _ in 0..10 {
+            let mut ct_clone = ct.clone();
+
+            ct_clone
+                .list
+                .ct_list
+                .ct_list
+                .get_mut_container()
+                .fill_with(|| rng.gen::<u64>());
+
+            assert!(ct_clone.is_conformant(&params));
+
+            let mut ct_clone_expanded = ct_clone.expand();
+
+            for i in &mut ct_clone_expanded {
+                *i += i.clone();
+            }
```

Comment:
Is this just a check that add operations work ok ? if so a small comment is welcome, I was a bit confused

Reply:
Added comment